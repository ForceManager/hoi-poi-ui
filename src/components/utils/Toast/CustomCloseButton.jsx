import React, { memo, useCallback } from 'react';
import PropTypes from 'prop-types';
import Icon from '../../general/Icon';

const CustomCloseButton = memo(({ properties, closeToast }) => {
    const handleCloseToast = useCallback(
        (e) => {
            e.preventDefault();
            e.stopPropagation();
            closeToast();
        },
        [closeToast],
    );

    if (!properties.closeButton && !properties.useDefaultCloseButton) return null;
    let classes = ['HoiPoi__ToastCloseButton'];
    if (properties.closeButtonClassName) classes.push(properties.closeButtonClassName);

    return (
        <>
            {properties.useDefaultCloseButton && (
                <Icon name="close" size="large" onClick={handleCloseToast} />
            )}
            {!properties.useDefaultCloseButton && (
                <div className={classes.join(' ')} onClick={handleCloseToast}>
                    {properties.closeButton}
                </div>
            )}
        </>
    );
});

CustomCloseButton.propTypes = {
    customCloseButton: PropTypes.element,
    closeToast: PropTypes.func,
};

export default CustomCloseButton;
