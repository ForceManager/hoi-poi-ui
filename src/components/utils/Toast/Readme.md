Default Toast:

```jsx
import { Button, Input, RadioGroup, toast } from 'hoi-poi-ui';

const state = {
    title: 'Lorem',
    text: 'Lorem ipsum',
    type: 'info',
    position: 'top-right',
    autoClose: 2000,
    newestOnTop: false,
    closeOnClick: true,
    ...state,
};

let radioOptions = [
    {
        label: 'Info',
        value: 'info',
    },
    {
        label: 'Success',
        value: 'success',
    },
    {
        label: 'Error',
        value: 'error',
    },
];

let positionOptions = [
    {
        label: 'top-left',
        value: 'top-left',
    },
    {
        label: 'top-right',
        value: 'top-right',
    },
    {
        label: 'bottom-left',
        value: 'bottom-left',
    },
    {
        label: 'bottom-right',
        value: 'bottom-right',
    },
];

let autoCloseOptions = [
    {
        label: 'false',
        value: false,
    },
    {
        label: '2000',
        value: 2000,
    },
    {
        label: '4000',
        value: 4000,
    },
    {
        label: '6000',
        value: 6000,
    },
];

let newestOnTopOptions = [
    {
        label: 'false',
        value: false,
    },
    {
        label: 'true',
        value: true,
    },
];

let closeOnClickOptions = [
    {
        label: 'false',
        value: false,
    },
    {
        label: 'true',
        value: true,
    },
];

let onChange = (value) => setState({ type: value });
let onChangeTitle = (e) => {
    setState({ title: e && e.target ? e.target.value : '' });
};
let onChangeText = (e) => {
    setState({ text: e && e.target ? e.target.value : '' });
};
let onChangePosition = (value) => setState({ position: value });
let onChangeAutoClose = (value) => setState({ autoClose: value });
let onChangeNewestOnTop = (value) => setState({ newestOnTop: !state.newestOnTop });
let onChangeCloseOnClick = (value) => setState({ closeOnClick: !state.closeOnClick });

<div>
    <RadioGroup label="Toast type" options={radioOptions} onChange={onChange} value={state.type} />
    <Input
        label="Toast title"
        placeholder="Type here"
        value={state.title}
        onChange={onChangeTitle}
    />
    <Input label="Toast test" placeholder="Type here" value={state.text} onChange={onChangeText} />
    <RadioGroup
        label="position"
        options={positionOptions}
        onChange={onChangePosition}
        value={state.position}
    />
    <RadioGroup
        label="autoClose"
        options={autoCloseOptions}
        onChange={onChangeAutoClose}
        value={state.autoClose}
    />
    <RadioGroup
        label="closeOnClick"
        options={closeOnClickOptions}
        onChange={onChangeCloseOnClick}
        value={state.closeOnClick}
    />
    <RadioGroup
        label="newestOnTop"
        options={newestOnTopOptions}
        onChange={onChangeNewestOnTop}
        value={state.newestOnTop}
    />
    <Toast position={state.position} autoClose={state.autoClose} newestOnTop={state.newestOnTop} />
    <br />
    <Button
        color="primary"
        onClick={() =>
            toast({
                type: state.type,
                text: state.text,
                title: state.title,
            })
        }
    >
        Show Toast
    </Button>
    <br />
    <br />
    <Button
        color="primary"
        onClick={() =>
            toast({
                content: <div className="custom-content">Component as content</div>,
                closeButton: <div style={{ color: 'white' }}>close</div>,
                closeButtonClassName: 'custom-closeButton-className',
                position: state.position,
                autoClose: state.autoClose,
                closeOnClick: state.closeOnClick,
                className: 'custom-one',
            })
        }
    >
        Show Custom Toast
    </Button>
</div>;
```

### Component tree

---

-   [root](https://github.com/fkhadra/react-toastify)
