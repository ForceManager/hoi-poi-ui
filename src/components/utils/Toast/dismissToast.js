import { toast } from 'react-toastify';

const dismissToast = (id) => {
    if (!id) return toast.dismiss();
    return toast.dismiss(id);
};

export default dismissToast;
