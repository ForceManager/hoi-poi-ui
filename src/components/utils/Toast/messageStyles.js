export default (theme) => ({
    root: {
        ...theme.typography.subtitle,
        fontWeight: '500',
        lineHeight: '20px',
    },
    title: {
        color: theme.colors.primary.white,
        display: 'block',
    },
    message: {
        ...theme.typography.caption,
        color: theme.colors.primary.white,
        display: 'block',
        whiteSpace: 'pre-wrap',
    },
    success: {
        '&$title': {
            color: theme.colors.green500,
        },
    },
    error: {
        '&$title': {
            color: theme.colors.red500,
        },
    },
    warning: {
        '&$title': {
            color: theme.colors.yellow500,
        },
    },
});
