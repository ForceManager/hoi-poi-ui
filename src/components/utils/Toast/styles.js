const timingFunction = {
    animationTimingFunction: 'cubic-bezier(.16, .37, .41, .99)',
};

export default (theme) => ({
    '@global': {
        '.Toastify__toast-container': {
            zIndex: 9999,
            webkitTransform: 'translate3d(0, 0, 9999px)',
            position: 'fixed',
            padding: 4,
            width: 350,
            boxSizing: 'border-box',
        },
        '.Toastify__toast-container--top-left': {
            top: '1em',
            left: '1em',
        },
        '.Toastify__toast-container--top-center': {
            top: '1em',
            left: '50%',
            transform: 'translateX(-50%)',
        },
        '.Toastify__toast-container--top-right': {
            top: '1em',
            right: '1em',
        },
        '.Toastify__toast-container--bottom-left': {
            bottom: '1em',
            left: '1em',
        },
        '.Toastify__toast-container--bottom-center': {
            bottom: '1em',
            left: '50%',
            transform: 'translateX(-50%)',
        },
        '.Toastify__toast-container--bottom-right': {
            bottom: '1em',
            right: '1em',
        },
        '@media only screen and (max-width : 480px)': {
            '.Toastify__toast-container': {
                width: '100vw',
                padding: '0',
                left: '0',
                margin: '0',
            },
            '.Toastify__toast-container--top-left, .Toastify__toast-container--top-center, .Toastify__toast-container--top-right':
                {
                    top: '0',
                    transform: 'translateX(0)',
                },
            '.Toastify__toast-container--bottom-left, .Toastify__toast-container--bottom-center, .Toastify__toast-container--bottom-right':
                {
                    bottom: '0',
                    transform: 'translateX(0)',
                },
            '.Toastify__toast-container--rtl': {
                right: '0',
                left: 'initial',
            },
            '.Toastify__toast': {
                marginBottom: '0',
                borderRadius: '0',
            },
        },
        '.Toastify__toast': {
            position: 'relative',
            minHeight: 64,
            boxSizing: 'border-box',
            marginBottom: '1rem',
            padding: '12px 16px',
            borderRadius: 8,
            display: 'flex',
            fallbacks: [
                {
                    display: '-ms-flexbox',
                },
            ],
            msFlexPack: 'justify',
            justifyContent: 'space-between',
            maxHeight: 800,
            overflow: 'hidden',
            cursor: 'pointer',
            direction: 'ltr',
            background: theme.colors.textLight.primary,
        },
        '.Toastify__toast--rtl': {
            direction: 'rtl',
        },
        '.Toastify__toast-body': {
            margin: 'auto 0',
            msFlex: '1 1 auto',
            flex: '1',
            ...theme.typography.body,
            color: theme.colors.primary.white,
        },
        '.Toastify__toast-icon': {
            webkitMarginEnd: 10,
            marginInlineEnd: 10,
            width: 20,
            msFlexNegative: '0',
            flexShrink: '0',
            display: 'flex',
            fallbacks: [
                {
                    display: '-ms-flexbox',
                },
            ],
        },
        '.Toastify--animate': {
            animationFillMode: 'both',
            animationDuration: '0.7s',
        },
        '.Toastify--animate-icon': {
            animationFillMode: 'both',
            animationDuration: '0.3s',
        },
        '.Toastify__toast--default': {},
        '.Toastify__toast--info': {},
        '.Toastify__toast--success': {
            borderLeft: `8px solid ${theme.colors.green500}`,
        },
        '.Toastify__toast--error': {
            borderLeft: `8px solid ${theme.colors.semantic.negative500}`,
        },
        '.Toastify__toast--warning': {
            borderLeft: `8px solid ${theme.colors.yellow500}`,
        },
        '.Toastify__progress-bar--info': {
            background: theme.colors.green500,
        },
        '.Toastify__progress-bar--success': {
            background: theme.colors.green500,
        },
        '.Toastify__progress-bar--warning': {
            background: theme.colors.yellow500,
        },
        '.Toastify__progress-bar--error': {
            background: theme.colors.semantic.negative500,
        },
        '@keyframes Toastify__trackProgress': {
            '0%': {
                transform: 'scaleX(1)',
            },
            '100%': {
                transform: 'scaleX(0)',
            },
        },
        '.Toastify__progress-bar': {
            position: 'absolute',
            bottom: '0',
            left: '0',
            width: '100%',
            height: 0,
            zIndex: 9999,
            opacity: '0.7',
            transformOrigin: 'left',
            display: 'none',
        },
        '.Toastify__progress-bar--animated': {
            animation: 'Toastify__trackProgress linear 1 forwards',
        },
        '.Toastify__progress-bar--controlled': {
            transition: 'transform 0.2s',
        },
        '.Toastify__progress-bar--rtl': {
            right: '0',
            left: 'initial',
            transformOrigin: 'right',
        },
        '.Toastify__spinner': {
            width: 20,
            height: 20,
            boxSizing: 'border-box',
            border: '2px solid',
            borderRadius: '100%',
            borderColor: '#e0e0e0',
            borderRightColor: '#616161',
            animation: 'Toastify__spin 0.65s linear infinite',
        },
        '@keyframes Toastify__bounceInRight': {
            'from, 90%, to': {
                ...timingFunction,
            },
            from: {
                opacity: 0,
                transform: 'translate3d(3000px, 0, 0)',
            },

            '90%': {
                transform: 'translate3d(-15px, 0, 0)',
            },
            to: {
                opacity: 1,
                transform: 'none',
            },
        },
        '@keyframes Toastify__bounceOutRight': {
            from: {
                opacity: 1,
                transform: 'none',
            },
            to: {
                opacity: 0,
                transform: 'translate3d(1000px, 0, 0)',
            },
        },
        '@keyframes Toastify__bounceInLeft': {
            'from, 90%, to': {
                ...timingFunction,
            },
            from: {
                opacity: 0,
                transform: 'translate3d(-3000px, 0, 0)',
            },
            '90%': {
                transform: 'translate3d(15px, 0, 0)',
            },
            to: {
                opacity: 1,
                transform: 'none',
            },
        },
        '@keyframes Toastify__bounceOutLeft': {
            from: {
                opacity: 1,
                transform: 'none',
            },
            to: {
                opacity: 0,
                transform: 'translate3d(-1000px, 0, 0)',
            },
        },
        '@keyframes Toastify__bounceInUp': {
            'from, 90%, to': {
                ...timingFunction,
            },
            from: {
                opacity: 0,
                transform: 'translate3d(0, 3000px, 0)',
            },
            '90%': {
                transform: 'translate3d(0, 15px, 0)',
            },
            to: {
                opacity: 1,
                transform: 'none',
            },
        },
        '@keyframes Toastify__bounceOutUp': {
            from: {
                opacity: 1,
                transform: 'none',
            },
            to: {
                opacity: 0,
                transform: 'translate3d(0, 3000px, 0)',
            },
        },
        '@keyframes Toastify__bounceInDown': {
            'from, 90%, to': {
                ...timingFunction,
            },
            from: {
                opacity: 0,
                transform: 'translate3d(0, -3000px, 0)',
            },
            '90%': {
                transform: 'translate(0, -15px, 0)',
            },
            to: {
                opacity: 1,
                transform: 'none',
            },
        },
        '@keyframes Toastify__bounceOutDown': {
            from: {
                opacity: 1,
                transform: 'none',
            },
            to: {
                opacity: 0,
                transform: 'translate3d(0, 15px, 0)',
            },
        },
        '.Toastify__bounce-enter--top-left, .Toastify__bounce-enter--bottom-left': {
            animationName: 'Toastify__bounceInLeft',
        },
        '.Toastify__bounce-enter--top-right, .Toastify__bounce-enter--bottom-right': {
            animationName: 'Toastify__bounceInRight',
        },
        '.Toastify__bounce-enter--top-center': {
            animationName: 'Toastify__bounceInDown',
        },
        '.Toastify__bounce-enter--bottom-center': {
            animationName: 'Toastify__bounceInUp',
        },
        '.Toastify__bounce-exit--top-left, .Toastify__bounce-exit--bottom-left': {
            animationName: 'Toastify__bounceOutLeft',
        },
        '.Toastify__bounce-exit--top-right, .Toastify__bounce-exit--bottom-right': {
            animationName: 'Toastify__bounceOutRight',
        },
        '.Toastify__bounce-exit--top-center': {
            animationName: 'Toastify__bounceOutUp',
        },
        '.Toastify__bounce-exit--bottom-center': {
            animationName: 'Toastify__bounceOutDown',
        },
        '@keyframes Toastify__zoomIn': {
            from: {
                opacity: '0',
                transform: 'scale3d(0.3, 0.3, 0.3)',
            },
            '50%': {
                opacity: '1',
            },
        },
        '@keyframes Toastify__zoomOut': {
            from: {
                opacity: '1',
            },
            '50%': {
                opacity: '0',
                transform: 'scale3d(0.3, 0.3, 0.3)',
            },
            to: {
                opacity: '0',
            },
        },
        '.Toastify__zoom-enter': {
            animationName: 'Toastify__zoomIn',
        },
        '.Toastify__zoom-exit': {
            animationName: 'Toastify__zoomOut',
        },
        '@keyframes Toastify__flipIn': {
            from: {
                transform: 'perspective(400px) rotate3d(1, 0, 0, 90deg)',
                animationTimingFunction: 'ease-in',
                opacity: '0',
            },
            '40%': {
                transform: 'perspective(400px) rotate3d(1, 0, 0, -20deg)',
                animationTimingFunction: 'ease-in',
            },
            '60%': {
                transform: 'perspective(400px) rotate3d(1, 0, 0, 10deg)',
                opacity: '1',
            },
            '80%': {
                transform: 'perspective(400px) rotate3d(1, 0, 0, -5deg)',
            },
            to: {
                transform: 'perspective(400px)',
            },
        },
        '@keyframes Toastify__flipOut': {
            from: {
                transform: 'perspective(400px)',
            },
            '30%': {
                transform: 'perspective(400px) rotate3d(1, 0, 0, -20deg)',
                opacity: '1',
            },
            to: {
                transform: 'perspective(400px) rotate3d(1, 0, 0, 90deg)',
                opacity: '0',
            },
        },
        '.Toastify__flip-enter': {
            animationName: 'Toastify__flipIn',
        },
        '.Toastify__flip-exit': {
            animationName: 'Toastify__flipOut',
        },
        '@keyframes Toastify__slideInRight': {
            from: {
                transform: 'translate3d(110%, 0, 0)',
                visibility: 'visible',
            },
            to: {
                transform: 'translate3d(0, 0, 0)',
            },
        },
        '@keyframes Toastify__slideInLeft': {
            from: {
                transform: 'translate3d(-110%, 0, 0)',
                visibility: 'visible',
            },
            to: {
                transform: 'translate3d(0, 0, 0)',
            },
        },
        '@keyframes Toastify__slideInUp': {
            from: {
                transform: 'translate3d(0, 110%, 0)',
                visibility: 'visible',
            },
            to: {
                transform: 'translate3d(0, 0, 0)',
            },
        },
        '@keyframes Toastify__slideInDown': {
            from: {
                transform: 'translate3d(0, -110%, 0)',
                visibility: 'visible',
            },
            to: {
                transform: 'translate3d(0, 0, 0)',
            },
        },
        '@keyframes Toastify__slideOutRight': {
            from: {
                transform: 'translate3d(0, 0, 0)',
            },
            to: {
                visibility: 'hidden',
                transform: 'translate3d(110%, 0, 0)',
            },
        },
        '@keyframes Toastify__slideOutLeft': {
            from: {
                transform: 'translate3d(0, 0, 0)',
            },
            to: {
                visibility: 'hidden',
                transform: 'translate3d(-110%, 0, 0)',
            },
        },
        '@keyframes Toastify__slideOutDown': {
            from: {
                transform: 'translate3d(0, 0, 0)',
            },
            to: {
                visibility: 'hidden',
                transform: 'translate3d(0, 500px, 0)',
            },
        },
        '@keyframes Toastify__slideOutUp': {
            from: {
                transform: 'translate3d(0, 0, 0)',
            },
            to: {
                visibility: 'hidden',
                transform: 'translate3d(0, -500px, 0)',
            },
        },
        '.Toastify__slide-enter--top-left, .Toastify__slide-enter--bottom-left': {
            animationName: 'Toastify__slideInLeft',
        },
        '.Toastify__slide-enter--top-right, .Toastify__slide-enter--bottom-right': {
            animationName: 'Toastify__slideInRight',
        },
        '.Toastify__slide-enter--top-center': {
            animationName: 'Toastify__slideInDown',
        },
        '.Toastify__slide-enter--bottom-center': {
            animationName: 'Toastify__slideInUp',
        },
        '.Toastify__slide-exit--top-left, .Toastify__slide-exit--bottom-left': {
            animationName: 'Toastify__slideOutLeft',
        },
        '.Toastify__slide-exit--top-right, .Toastify__slide-exit--bottom-right': {
            animationName: 'Toastify__slideOutRight',
        },
        '.Toastify__slide-exit--top-center': {
            animationName: 'Toastify__slideOutUp',
        },
        '.Toastify__slide-exit--bottom-center': {
            animationName: 'Toastify__slideOutDown',
        },
        '@keyframes Toastify__spin': {
            from: {
                transform: 'rotate(0deg)',
            },
            to: {
                transform: 'rotate(360deg)',
            },
        },
    },
});
