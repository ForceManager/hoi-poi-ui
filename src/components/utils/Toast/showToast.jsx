import React from 'react';
import { toast } from 'react-toastify';
import ToastMessage from './ToastMessage';
import CustomCloseButton from './CustomCloseButton';

const TOAST_TYPES = {
    success: toast.TYPE.SUCCESS,
    info: toast.TYPE.INFO,
    error: toast.TYPE.ERROR,
};

const showToast = (props) => {
    const {
        type = 'info',
        text = '',
        title = '',
        containerId,
        content,
        closeButton,
        closeButtonClassName,
        ...newProps
    } = props;

    if (content) {
        const properties = {
            closeButton,
            closeButtonClassName,
        };

        return toast(content, {
            containerId: containerId || 'hoi-poi-ui',
            closeButton: <CustomCloseButton properties={properties} />,
            ...newProps,
        });
    }

    const toastType = TOAST_TYPES[type];
    const messageProps = {
        title,
        text,
        type,
    };

    return toast(<ToastMessage {...messageProps} />, {
        type: toastType,
        containerId: containerId || 'hoi-poi-ui',
        ...newProps,
    });
};

export default showToast;
