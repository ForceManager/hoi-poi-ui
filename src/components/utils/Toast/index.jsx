import { toast } from 'react-toastify';
import Toast from './Toast';
import showToast from './showToast';

const dismiss = toast.dismiss;
export { Toast, showToast as toast, dismiss };
