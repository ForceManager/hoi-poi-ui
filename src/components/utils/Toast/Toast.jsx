import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'react-jss';
import classnames from 'classnames';
import { ToastContainer } from 'react-toastify';

import { getOverrides } from '../../../utils/overrides';
import styles from './styles';

function Toast({
    classes,
    className: classNameProp,
    overrides: overridesProp,
    containerId,
    ...props
}) {
    // Overrides
    const override = getOverrides(overridesProp, Toast.overrides);

    // Classes
    const rootClassName = classnames(classes.root, {}, classNameProp);

    const rootProps = {
        ...props,
        className: rootClassName,
        closeButton: false,
        hideProgressBar: true,
        autoClose: 4000,
        containerId: containerId || 'hoi-poi-ui',
    };

    return <ToastContainer {...rootProps} enableMultiContainer {...override['react-toastify']} />;
}

Toast.overrides = [];

Toast.defaultProps = {
    className: '',
    overrides: {},
};

Toast.propTypes = {
    className: PropTypes.string,
    overrides: PropTypes.object,
    containerId: PropTypes.any,
};

export default React.memo(withStyles(styles, { name: 'Toast' })(Toast));
