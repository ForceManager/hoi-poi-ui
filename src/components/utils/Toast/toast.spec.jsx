import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HoiPoiProvider } from '../../../utils/styles';
import { Toast } from '../../utils/Toast';

describe('Toast', () => {
    test('default match', () => {
        const wrapper = mount(
            <HoiPoiProvider>
                <Toast />
            </HoiPoiProvider>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
