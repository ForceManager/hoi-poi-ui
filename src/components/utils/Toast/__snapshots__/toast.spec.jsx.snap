// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast default match 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "aqua100": "#e4fafd",
          "aqua500": "#76bde8",
          "aqua900": "#16386f",
          "black": "#262B34",
          "blackLight": "#5f656b",
          "blackberry": "#2e2d4c",
          "blackberryDark": "#24243C",
          "blue100": "#cceeff",
          "blue200": "#99d8ff",
          "blue400": "#3fa4ff",
          "blue500": "#007aff",
          "blue600": "#005edb",
          "blue900": "#00237a",
          "disabledGrey": "#ced4d9",
          "green": "#86AD20",
          "green100": "#f4fad2",
          "green200": "#e7f6a6",
          "green400": "#aecd50",
          "green500": "#86ad20",
          "green600": "#6e9417",
          "green900": "#345306",
          "greenLight": "#e7f1d5",
          "grey": "#697782",
          "greyLight": "#f0f3f5",
          "greySoft": "#788590",
          "lines": "#dce3eb",
          "link": "#216db3",
          "linkDark": "#174a79",
          "neutral100": "#f9f9fa",
          "neutral200": "#f4f5f6",
          "neutral300": "#e9ecee",
          "neutral400": "#dfe2e5",
          "neutral500": "#d4d8dc",
          "neutral600": "#a9b1b9",
          "neutral700": "#7d8a96",
          "neutral800": "#526373",
          "neutral900": "#273c50",
          "neutralBase": "#ffffff",
          "orange100": "#fff1cc",
          "orange200": "#ffde99",
          "orange400": "#ffb13f",
          "orange500": "#ff8c00",
          "orange600": "#db6e00",
          "orange900": "#7a2c00",
          "overlay": "rgba(33,33,33,0.55)",
          "placeholders": "#e1e4e7",
          "placeholdersHover": "#d0d4d8",
          "primary": "#FF8C00",
          "primaryDark": "#C87600",
          "primaryLight": "#fef4e6",
          "primarySoft": "#ff9d26",
          "purple": "#AA5B9C",
          "purple100": "#fae3ed",
          "purple500": "#aa5b9c",
          "purple900": "#441151",
          "purpleLight": "#EEDEEB",
          "red": "#f04540",
          "red100": "#ffe5d5",
          "red200": "#ffc5ac",
          "red400": "#ff7863",
          "red500": "#ff3b30",
          "red600": "#db2328",
          "red900": "#7a0929",
          "redDark": "#C43833",
          "redLight": "#FAE7E7",
          "redSoft": "#FF544F",
          "secondary": "#00A8Be",
          "secondaryGrey": "#e4eeff",
          "secondaryLight": "#D6EEF2",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "secondarySoft": "#AADFE7",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "tableHeader": "#f6f7f9",
          "tableHeaderHover": "#EBECEF",
          "tableHover": "#FAFBFC",
          "text": Object {
            "black": "#3c4248",
            "danger": "white",
            "default": "#697782",
            "green": "#86AD20",
            "greySoft": "#788590",
            "primary": "white",
            "red": "#f04540",
            "secondary": "#00A8Be",
            "white": "#FFFFFF",
            "yellow": "#E9B200",
          },
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "turquoise100": "#c9fbf1",
          "turquoise500": "#00a8be",
          "turquoise900": "#00335b",
          "white": "#FFFFFF",
          "yellow": "#E9B200",
          "yellow100": "#fdf6cb",
          "yellow200": "#fcec97",
          "yellow400": "#f1cb3c",
          "yellow500": "#e9b200",
          "yellow600": "#c89400",
          "yellow900": "#6f4a00",
          "yellowDark": "#BE9100",
          "yellowLight": "#FCF7E5",
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": Object {
          "bigText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "lineHeight": 1.5,
          },
          "defaultText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 13,
            "letterSpacing": 0.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 13,
          "fontSizeSmall": 11,
          "largeText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "smallText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
        },
        "utils": Object {
          "scrollbar": Object {
            "&::-webkit-scrollbar": Object {
              "height": 17,
              "width": 17,
            },
            "&::-webkit-scrollbar-thumb": Object {
              "background": "#e1e4e7",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
            "&::-webkit-scrollbar-thumb:window-inactive": Object {
              "background": "#e1e4e7",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": Object {
              "minify": false,
            },
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.3.0",
        }
      }
    >
      <Toast
        autoClose={4000}
        className=""
        overrides={Object {}}
      >
        <ToastContainer
          autoClose={4000}
          className="HoiPoi__Toast__root-1-9"
          closeButton={false}
          closeButtonClassName=""
          closeOnClick={true}
          containerId="hoi-poi-ui"
          draggable={true}
          draggablePercent={80}
          enableMultiContainer={true}
          hideProgressBar={true}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          role="alert"
          rtl={false}
          transition={[Function]}
        >
          <div
            className="Toastify"
            id="hoi-poi-ui"
          />
        </ToastContainer>
      </Toast>
    </JssProvider>
  </ThemeProvider>
</Component>
`;
